src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -133,28 +172,29 @@ def get_disk_details(self, block_device, disk, lshw_disks):
             'subsystem': os.path.realpath(os.path.join(disk_sys_path, 'device/subsystem')).split('/')[-1],
         })
 
-        disk['sectorsize'] = self.logical_sector_size(block_device.sys_name)
-
-        type_path = os.path.join(disk_sys_path, 'queue/rotational')
-        if os.path.exists(type_path):
-            with open(type_path, 'r') as f:
-                disk['type'] = 'SSD' if f.read().strip() == '0' else 'HDD'
-        else:
-            self.middleware.logger.error(
-                'Unable to retrieve %r disk rotational details at %s', disk['name'], type_path
-            )
+        if device_path in lsblk_disks:
+            disk_data = lsblk_disks[device_path]
 
-        if device_path in lshw_disks:
-            disk_data = lshw_disks[device_path]
+            # get type of disk and rotational rate (if HDD)
+            disk['type'] = 'SSD' if not disk_data['rota'] else 'HDD'
             if disk['type'] == 'HDD':
-                disk['rotationrate'] = disk_data['rotationrate']
+                disk['rotationrate'] = self.get_rotational_rate(device_path)
 
+            # get model and serial
             disk['ident'] = disk['serial'] = disk_data.get('serial', '')
-            disk['size'] = disk['mediasize'] = int(disk_data['size']) if 'size' in disk_data else None
-            disk['descr'] = disk['model'] = disk_data.get('product')
+            disk['descr'] = disk['model'] = disk_data.get('model')
+
+            # get all relevant size attributes of disk
+            disk['sectorsize'] = disk_data['log-sec']
+            disk['size'] = disk_data['size']
```

Comment:
I think we should still set `mediasize` for backwards compatibility on consumers consuming that key. We do that for freebsd side as well.

Reply:
oops, missed a key. good catch, I'll fix.